### PR TITLE
Chaeng final week/ver2

### PR DESCRIPTION
**Pull request by Chaeng**
Updated reflection and refraction computation.

### CODE REVIEW

File: Final Project/Assets/RenderPipeline/NewComputeShader.compute [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿// Each #kernel tells which function to compile; you can have many kernels
```
Comment (achekery) [inline Final Project/Assets/RenderPipeline/NewComputeShader.compute:1]: what is new compute shader needed for?

File: Final Project/Assets/RenderPipeline/SecondaryRay/RefractRay.compute [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+     float invN = m_in/ m_out;
+     float cosThetaT = sqrt( 1 - ((invN * invN) * (1 - (cosThetaI * cosThetaI))));
+     float3 transDir = -((invN * V) + ((cosThetaT - (invN * cosThetaI)) * useNormal));
+     Ray transRay = CreateRay(curHit.position, -transDir);
+     return transRay;
+}
+
+#endif
+
+/*
```
Comment (achekery) [inline Final Project/Assets/RenderPipeline/SecondaryRay/RefractRay.compute:32]: we can delete the disabled code

File: Final Project/Assets/RenderPipeline/NewComputeShader.compute (hunk 1 above)
Comment (Chaeng) [reply Final Project/Assets/RenderPipeline/NewComputeShader.compute:1]: Please ignore this. This is a tmp shader for when the RayTracing crash. 
Joe has helped me created this to trick shader to run something. It won't hurt anyone's code.

File: Final Project/Assets/RenderPipeline/SecondaryRay/RefractRay.compute (hunk 2 above)
Comment (Chaeng) [reply Final Project/Assets/RenderPipeline/SecondaryRay/RefractRay.compute:32]: This is for my ref. to confirm the calculation.

Comment (achekery) [reply Final Project/Assets/RenderPipeline/SecondaryRay/RefractRay.compute:32]: This won't hurt any code

**Review by joeccleung — APPROVED** [review]

File: Final Project/Assets/RenderPipeline/SecondaryRay/GetTriangleNormal.compute [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿#ifndef GETTRIANGLENORMAL_COMPUTE
```
Comment (kaihuh) [inline Final Project/Assets/RenderPipeline/SecondaryRay/GetTriangleNormal.compute:1]: You can get normal directly from the triangle like 
_Triangles[index].normal
or you can get it from Rayhit.normal
You might don't need to use this fuction.

Comment (Chaeng) [reply Final Project/Assets/RenderPipeline/SecondaryRay/GetTriangleNormal.compute:1]: true. thanks!

**Pull request merged**